lib/spack/spack/compilers/xl.py

```diff
@@ -43,6 +43,14 @@ def cxx11_flag(self):
         else:
             return "-qlanglvl=extended0x"
 
+    @property
+    def c99_flag(self):
+        return '-std=c99'
+
+    @property
+    def c11_flag(self):
+        return '-std=c11'
+
```

Comment:
The logic here is much more complicated for XL based on when the switch to the clang front end happened:

| Ver | C99 | C11 |
| - | - | - |
| 10.1 | -qlanglvl=stdc99 | X |
| 12.1 | -qlanglvl=stdc99 | -qlanglvl=extc1x |
| 13.1.1 | -std=c99 | -qlanglvl=extc1x |
| 13.1.2 | -std=c99 | -std=c11 |

Reply:
See #11635